### PR TITLE
Add an optional emptyValue to SelectInput

### PR DESCRIPTION
**Pull request by edy**
#2234 introduced a breaking change for our application. This change adds an optional custom value which is used when allowEmpty is set.

Previously `null` was set as an empty value, now you can set null again.

### CODE REVIEW

**Comment by sghribi** [comment]
Thanks @edy for this PR!

Is it ok to be merged for 2.7.1?

**Comment by edy** [comment]
sure

File: packages/ra-ui-materialui/src/input/SelectInput.js [hunk 1]
```diff
@@ -155,7 +156,10 @@ export class SelectInput extends Component {
 
     addAllowEmpty = choices => {
         if (this.props.allowEmpty) {
-            return [<MenuItem value="" key="null" />, ...choices];
+            const value = this.props.hasOwnProperty('emptyValue')
+                ? this.props.emptyValue
```
Comment (djhi) [inline packages/ra-ui-materialui/src/input/SelectInput.js]: Using `defaultProps` might be better than this manual check

Comment (edy) [reply packages/ra-ui-materialui/src/input/SelectInput.js]: i agree. will change that later this day

**Comment by edy** [comment]
> Is it ok to be merged for 2.7.1?

would you merge this for 2.7.2?

**Comment by sghribi** [comment]
Hi @djhi @fzaninotto 

Could you review/validate this PR?

Thanks!

**Comment by djhi** [comment]
Sorry @edy, as it is a new prop, can you please target the `next` branch?

**Comment by edy** [comment]
Done.

Why isn't it possible to merge into master? All tests pass and the change is backward compatible too.

**Comment by djhi** [comment]
As I said in previous comment, this adds a new prop, hence a new feature. If we do merge it, it will be on the `next` branch for the `2.8` release (semver ?)

**Comment by edy** [comment]
also done

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**